webui/react/src/e2e/fixtures/api.project.fixture.ts

```diff
@@ -75,19 +55,17 @@ export class ApiProjectFixture {
     await expect
       .poll(
         async () => {
-          const projectResp = await (await this.startProjectRequest())
-            .deleteProject(id)
-            .catch(async function (error) {
-              const respBody = await streamConsumers.text(error.body);
-              if (error.status === 404) {
-                return { completed: true };
-              }
-              throw new Error(
-                `Delete Project Request failed. Status: ${error.status} Request: ${JSON.stringify(
-                  id,
-                )} Response: ${respBody}`,
-              );
-            });
+          const projectResp = await this.api.deleteProject(id).catch(async function (error) {
+            const respBody = await streamConsumers.text(error.body);
+            if (error.status === 404) {
+              return { completed: true };
+            }
```

Comment:
q: why is a 404 being treated as successful?

Reply:
Because the project deletion process runs asynchronously on the server, it looks like we're working around this by polling the delete endpoint until it either returns `{completed: true}` or the project isn't found, at which point we can assume it was deleted.